statistics/handle/lru_cache_test.go

```diff
@@ -0,0 +1,221 @@
+	indexMemoryUsage  = int64(4)
+)
+
+// each column and index consumes 4 bytes memory
+func newMockStatisticsTable(columns int, indices int) *statistics.Table {
+	t := &statistics.Table{}
+	t.Columns = make(map[int64]*statistics.Column)
+	t.Indices = make(map[int64]*statistics.Index)
+	for i := 1; i <= columns; i++ {
+		t.Columns[int64(i)] = &statistics.Column{
+			Info:     &model.ColumnInfo{ID: int64(i)},
+			CMSketch: statistics.NewCMSketch(1, 1),
+		}
+	}
+	for i := 1; i <= indices; i++ {
+		t.Indices[int64(i)] = &statistics.Index{
+			Info:     &model.IndexInfo{ID: int64(i)},
+			CMSketch: statistics.NewCMSketch(1, 1),
+		}
+	}
+	return t
+}
+
+func mockTableAppendColumn(t *statistics.Table) {
+	index := int64(len(t.Columns) + 1)
+	t.Columns[index] = &statistics.Column{
+		Info:     &model.ColumnInfo{ID: index},
+		CMSketch: statistics.NewCMSketch(1, 1),
+	}
+}
+
+func mockTableAppendIndex(t *statistics.Table) {
+	index := int64(len(t.Columns) + 1)
+	t.Indices[index] = &statistics.Index{
+		Info:     &model.IndexInfo{ID: index},
+		CMSketch: statistics.NewCMSketch(1, 1),
+	}
+}
+
+func TestLRUPutGetDel(t *testing.T) {
+	capacity := int64(100)
+	lru := newInternalLRUCache(capacity)
+	require.Equal(t, capacity, lru.capacity)
+	mockTable := newMockStatisticsTable(1, 1)
+	mockTableID := int64(1)
+	lru.Put(mockTableID, mockTable)
+	v, ok := lru.Get(mockTableID)
+	require.True(t, ok)
+	require.Equal(t, mockTable, v)
+	lru.Del(mockTableID)
+	v, ok = lru.Get(mockTableID)
+	require.False(t, ok)
+	require.Nil(t, v)
+}
+
+func TestLRUEvict(t *testing.T) {
+	capacity := int64(12)
+	lru := newInternalLRUCache(capacity)
+	t1 := newMockStatisticsTable(2, 1)
+	require.Equal(t, t1.MemoryUsage().TotalMemUsage, 2*columnMemoryUsage+1*indexMemoryUsage)
+	require.Equal(t, t1.MemoryUsage().TotalColTrackingMemUsage(), 2*columnMemoryUsage)
+	lru.Put(int64(1), t1)
+	require.Equal(t, lru.totalCost, t1.MemoryUsage().TotalMemUsage)
+	require.Equal(t, lru.trackingCost, t1.MemoryUsage().TotalColTrackingMemUsage())
+	t2 := newMockStatisticsTable(1, 2)
+	lru.Put(int64(2), t2)
+	require.Equal(t, lru.totalCost, t1.MemoryUsage().TotalMemUsage+t2.MemoryUsage().TotalMemUsage)
+	require.Equal(t, lru.trackingCost, t1.MemoryUsage().TotalColTrackingMemUsage()+t2.MemoryUsage().TotalColTrackingMemUsage())
+
+	// Put t3, a column CMSketch of t1 should be evicted
+	t3 := newMockStatisticsTable(1, 3)
+	lru.Put(int64(3), t3)
+	require.Equal(t, lru.Len(), 3)
+	require.Equal(t, t1.MemoryUsage().TotalColTrackingMemUsage(), columnMemoryUsage)
+	require.Equal(t, lru.totalCost, 3*columnMemoryUsage+6*indexMemoryUsage)
+	require.Equal(t, lru.trackingCost, 3*columnMemoryUsage)
+	require.Equal(t, lru.totalCost, t1.MemoryUsage().TotalMemUsage+t2.MemoryUsage().TotalMemUsage+t3.MemoryUsage().TotalMemUsage)
+	require.Equal(t, lru.trackingCost, t1.MemoryUsage().TotalColTrackingMemUsage()+
+		t2.MemoryUsage().TotalColTrackingMemUsage()+t3.MemoryUsage().TotalColTrackingMemUsage())
```

Comment:
Can we check `t1.MemoryUsage().TotalColTrackingMemUsage() ==  columnMemoryUsage` here? In this way we can ensure only one column CMSketch remains for `t1`.

Reply:
It is already checked in line 102